src/libraries/System.Private.CoreLib/src/System/IO/TextReader.cs

```diff
@@ -203,18 +203,26 @@ public virtual int ReadBlock(Span<char> buffer)
         }
 
         #region Task based Async APIs
-        public virtual Task<string?> ReadLineAsync() =>
+        public virtual Task<string?> ReadLineAsync() => ReadLineCoreAsync(default);
+
+        public virtual ValueTask<string?> ReadLineAsync(CancellationToken cancellationToken) =>
+            new ValueTask<string?>(ReadLineCoreAsync(cancellationToken));
+
+        private Task<string?> ReadLineCoreAsync(CancellationToken cancellationToken) =>
             Task<string?>.Factory.StartNew(static state => ((TextReader)state!).ReadLine(), this,
-                CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
+                cancellationToken, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
 
-        public virtual async Task<string> ReadToEndAsync()
+        public virtual Task<string> ReadToEndAsync() =>
+            ReadToEndAsync(default);
```

Comment:
nit: one line should be fine

```suggestion
        public virtual Task<string> ReadToEndAsync() => ReadToEndAsync(default);
```